### PR TITLE
Update project's dependencies to latest compatible versions

### PR DESCRIPTION
**Pull request by oleiade**
In this PR, we update the project's dependencies to their latest compatible versions. This was overdue.
As we update to a much more recent version of k6, we also take this opportunity to switch to using WebCrypto's `crypto.RandomUUID` in the secrets manager too.

### CODE REVIEW

File: src/internal/secrets-manager.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { JSONArray, JSONObject } from 'k6'
 import http, { RefinedResponse, ResponseType } from 'k6/http'
+import { crypto } from 'k6/experimental/webcrypto'
```
Comment (codebien) [inline src/internal/secrets-manager.ts:3]: Just as a case report. We don't expect to remove the WebCrypto module, but here we are building a product on top of an experimental feature.

**Review by codebien — APPROVED** [review]
LGTM

File: src/internal/secrets-manager.ts (hunk 1 above)
Comment (joanlopez) [reply src/internal/secrets-manager.ts:3]: Yeah, I agree that technically we shouldn't do so unless we consider `k6-jslib-aws` also experimental. With this, if k6 users start using the next release of this library and, at some point, we move `webcrypto` to _stable_, they'll start getting warnings, that aren't directly produced by their code.

To be fair, I don't think there's such a big deal here, but definitely something that worths agreement within the team, and to set a precedent (guidelines) for future cases.

Comment (codebien) [reply src/internal/secrets-manager.ts:3]: > they'll start getting warnings

This is a good catch. It sounds relevant enough to roll back the change. Considering that we already have a working alternative to it.

Comment (oleiade) [reply src/internal/secrets-manager.ts]: Good point folks 👍🏻 I switched to k6-utils' `uuidv4` in the meantime.

File: src/internal/secrets-manager.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { JSONArray, JSONObject } from 'k6'
 import http, { RefinedResponse, ResponseType } from 'k6/http'
+import { uuidv4 } from 'https://jslib.k6.io/k6-utils/1.4.0/index.js'
 
-import { v4 as uuidv4 } from 'uuid'
```
Comment (codebien) [inline src/internal/secrets-manager.ts:4]: I'm not sure I'm getting what's the problem with this dependency. Can we just keep it without applying the change?

Comment (oleiade) [reply src/internal/secrets-manager.ts:4]: Apologies, I should have been more explicit about this specifically 🙇🏻  

So far, k6-jslib-aws relied on the uuid package to generate uuids. We were using version 3.4.0, which is quite old, but had the advantage of not relying on node's `crypto.getRandomValues` API. This version is now deprecated, and the newer versions (from `7.0.0`, to `10.0.0`) all rely on the node API exclusively (as far as I can tell), meaning we can't rely on it in a k6 context anymore.

Hence, switching to k6-utils instead. (Which coincidentally makes me realize there is a `secure` flag to activate to use a cryptographic random source instead)

**Review by codebien — APPROVED** [review]

File: src/internal/secrets-manager.ts (hunk 2 above)
Comment (joanlopez) [reply src/internal/secrets-manager.ts:4]: Thanks for the explanation @oleiade, all clear now! Looks great! 👍🏻

**Review by joanlopez — APPROVED** [review]

**Pull request merged**